src/node-saml/saml.ts

```diff
@@ -1404,6 +1426,29 @@ class SAML {
 
     throw new Error("Invalid key");
   }
+
+  /**
+   * Process max age assertion and use it if it is more restrictive than the NotOnOrAfter age
+   * assertion received in the SAMLResponse.
+   *
+   * @param maxAssertionAgeMs Max time after IssueInstant that we will accept assertion, in Ms.
+   * @param notOnOrAfter Expiration provided in response.
+   * @param issueInstant Time when response was issued.
+   * @returns {*} The expiration time to be used, in Ms.
+   */
+  private processMaxAgeAssertionTime(
+    maxAssertionAgeMs: number,
+    notOnOrAfter: string,
+    issueInstant: string
+  ): number {
+    const notOnOrAfterMs = Date.parse(notOnOrAfter);
```

Comment:
I feel like there should be some checks for `NaN` here (along with a test or two). A comparison with `NaN` will return `false` every time. Thus, if `issueInstant` parses out to `NaN`, then `notOnOrAfterMs` will be returned every time. Likewise if `notOnOrAfter` parses out to `NaN`, then `NaN` will be returned every time.